### PR TITLE
Add API service to Docker

### PR DESCRIPTION
**Pull request by schreiaj**
## Description
Builds a live reloading instance of the api service for development. You will need to build the java code (`./gradlew build`) but then the server will pick up the changes.

## Screenshots (if applicable)
N/A

## Related Issues
N/A - Dev environment simplification

## Checklist
- [x] The title of this PR is descriptive and concise.
- [x] My changes follow the style guidelines of this project.
- [x] I have added or updated test cases to cover my changes.
- [x] I've let the team know about this PR by linking it in the review channel

### CODE REVIEW

File: dev-env.yaml [hunk 1]
```diff
@@ -18,6 +18,9 @@ services:
       - "8080:8080"
     volumes:
       - ./frontend:/app
+    depends_on:
+      - api
+      - ocr
```
Comment (arinkulshi-skylight) [inline dev-env.yaml:23]: Would it makes sense to add `depends_on:  - db`

**Review by arinkulshi-skylight — APPROVED** [review]

File: dev-env.yaml (hunk 1 above)
Comment (schreiaj) [reply dev-env.yaml:23]: Maybe? But I would argue that the frontend doesn't know anything about the DB.

**Comment by schreiaj** [comment]
> Are we just adding this to dev for now and the main dockerfile later on?

Yes, this is just for the dev environment for now.

File: dev-env.yaml (hunk 1 above)
Comment (arinkulshi-skylight) [reply dev-env.yaml:23]: I sort of imagined it as DB then API then, OCR then frontend but I guess it is ok to keep the logic separate.

**Pull request merged**